components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/dao/RequestObjectDAOImpl.java

```diff
@@ -279,6 +279,49 @@ private void insertRequestObjectClaimValues(int requestObjectClaimId, List<Strin
         return essentialClaims;
     }
 
+    /**
+     * Retrieve essential claims for the id token and user info endpoint.
+     *
+     * @param sessionDataKey session Data Key
+     * @param isUserInfo     return true if the claims are requested from user info end point.
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    @Override
+    public List<RequestedClaim> getRequestedClaimsBySessionDataKey(String sessionDataKey, boolean isUserInfo)
```

Comment:
Do we really need to get requested claim by session data key ? As per the discussion we had with @ayshsandu we only need a mechanism to get the latest entry of requested claims stored against token. @ayshsandu can you please add some insight here ?

Reply:
Do you see any harm in keeping this? As of now, to get rid of the difficulty we face in 'code id_token' hybrid flow, we are directly hitting this query against the database than going through the service layer. I suppose this is better than that.